src/Storage/AbstractSessionArrayStorage.php

```diff
@@ -207,10 +195,8 @@ public function getIterator()
      * Load session object from an existing array
      *
      * Ensures $_SESSION is set to an instance of the object when complete.
-     *
-     * @return SessionStorage
      */
-    public function fromArray(array $array)
+    public function fromArray(array $array): self
```

Comment:
```suggestion
    public function fromArray(array $array): static
```

This should probably be changed in the interface too. Returning `static` preserves your type, i.e. `$type = SomeConcrete::fromArray(['foo'])` will mean `$type instanceof SomeConcrete` rather than `SomeInterface`